core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -342,7 +342,7 @@
     /**
      * The Jenkins instance startup type i.e. NEW, UPGRADE etc
      */
-    private String installStateName;
+    private transient String installStateName;
```

Comment:
Probably a regression when you restart Jenkins during the setup wizard (can happen when installing a plugin that cannot be dynamically loaded IIRC).

Reply:
yup https://github.com/jenkinsci/jenkins/pull/5283#issuecomment-778048669 and https://github.com/jenkinsci/jenkins/pull/5283#issuecomment-778049697

Reply:
I think this could be limited to happening only while the setup wizard is active.

Reply:
I was wondering about either doing this based off the initial admin password file being present or creating a new file, or do you have a different thought?